builtin/credential/approle/path_role.go

```diff
@@ -2355,9 +2375,38 @@ func (b *backend) handleRoleSecretIDCommon(ctx context.Context, req *logical.Req
 		return nil, err
 	}
 
+	var numUses int
+	// Check whether or not num_uses is defined, otherwise fallback to secret_id_num_uses
+	if numUsesRaw, ok := data.GetOk("num_uses"); ok {
+		numUses = numUsesRaw.(int)
+		if numUses < 0 {
+			return logical.ErrorResponse("num_uses cannot be negative"), nil
+		}
+
+		// If the role's secret_id_num_uses is higher than the specified num_uses, throw an error rather than implicitly overriding
+		if (numUses == 0 && role.SecretIDNumUses > 0) || (role.SecretIDNumUses > 0 && numUses > role.SecretIDNumUses) {
+			return logical.ErrorResponse("num_uses cannot be higher than the role's secret_id_num_uses"), nil
+		}
+	} else {
+		numUses = role.SecretIDNumUses
+	}
+
+	var ttl time.Duration
+	// Check whether or not ttl is defined, otherwise fallback to secret_id_ttl
+	if ttlRaw, ok := data.GetOk("ttl"); ok {
+		ttl = time.Second * time.Duration(ttlRaw.(int))
```

Comment:
This piece of code was duplicated from the current approach taken to calculate the Role's Secret ID's ttl: https://github.com/hashicorp/vault/blob/22346ba24980073f8243aa5a1ee35dc18427da49/builtin/credential/approle/path_role.go#L966

Seeing that this is no longer valid, do we want to change this behaviour as well? The code was duplicated only with the intention to not change the way TTL is calculated for the SecretID.

Reply:
We don't want to change the behavior of the existing API, at least not in this PR. It's odd that the implementation you point out doesn't match the docs (which says it takes either a duration string or an integer). Regardless, since this is a new API, we should use `parseutil.ParseDurationSecond()`, since that will handle both duration strings and integers transparently.

Reply:
I would argue that this isn't a new API. We are currently modifying an already existing API endpoint with additional new fields. It would be odd to change this now as it defaults to the Role's Secret ID Configuration, which is calculated using the invalid approach.
How about we immediatelly create a new issue and link it after this PR?

Reply:
You're right, I misspoke. This is not a new API. It's a new field that we're adding to an existing API. I'm arguing that since it's a new field, the value we extract from it should be parsed using the more flexible approach, that allows for either integers or duration strings. The fact that the existing TTL field doesn't do that is a bug, as far as I can tell, especially since the documentation indicates that it should. It doesn't make sense to me to merge this PR with a bug in it, when we can correct it before it's merged.

Reply:
While I agree, now we could have a potential mismatch in the way that TTL is parsed and returned, just to point that out. Since it's obviously not a good thing to leave bugs in, I will create an issue for the existing TTL fields. In the mean time I will patch the behaviour of the new field and we can go from there!

Reply:
> This is a major change. We should be careful about this.

I disagree that it's a major change, since I'm not advocating for changing the existing behavior of anything. Everything that existed before this PR will continue to work the same way it always has. All we're doing is determining how to parse a brand new field that no one has used before.